### PR TITLE
Implementaion of ext_proc filter disable immediate response

### PR DESCRIPTION
**Pull request by yanjunxiang-google**
This is implementation to address issue: https://github.com/envoyproxy/envoy/issues/28698.

It's a follow up PR of https://github.com/envoyproxy/envoy/pull/28907

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: changelogs/current.yaml [hunk 1]
```diff
@@ -46,6 +46,10 @@ new_features:
 - area: extension_discovery_service
   change: |
     added ECDS support for :ref:` downstream network filters<envoy_v3_api_field_config.listener.v3.Filter.config_discovery>`.
+- area: ext_proc
+  change: |
+    added disable_immediate_response config API to ignore immediate response from the external processing server.
```
Comment (htuch) [inline changelogs/current.yaml:51]: Please make this a `:ref:`

File: source/extensions/filters/http/ext_proc/ext_proc.cc [hunk 2]
```diff
@@ -653,19 +653,26 @@ void Filter::onReceiveMessage(std::unique_ptr<ProcessingResponse>&& r) {
...
-    // avoided.
-    logGrpcStreamInfo();
-    processing_complete_ = true;
-    closeStream();
-    onFinishProcessorCalls(Grpc::Status::Ok);
-    sendImmediateResponse(response->immediate_response());
-    processing_status = absl::OkStatus();
+    if (config_->disableImmediateResponse()) {
+      ENVOY_LOG(debug,
+                "Filter has diable immediate response configured. Treat it as spurious response");
```
Comment (htuch) [inline source/extensions/filters/http/ext_proc/ext_proc.cc:658]: Nit: s/diable/disable/

**Review by htuch — CHANGES_REQUESTED** [review]
LGTM modulo two nits.

File: test/extensions/filters/http/ext_proc/filter_test.cc [hunk 3]
```diff
@@ -701,6 +701,40 @@ TEST_F(HttpFilterTest, PostAndRespondImmediately) {
...
+  immediate_response->set_body("Bad request");
+  immediate_response->set_details("Got a bad request");
+  auto* immediate_headers = immediate_response->mutable_headers();
+  auto* hdr1 = immediate_headers->add_set_headers();
+  hdr1->mutable_header()->set_key("content-type");
+  hdr1->mutable_header()->set_value("text/plain");
+  stream_callbacks_->onReceiveMessage(std::move(resp1));
+
+  Buffer::OwnedImpl req_data("foo");
+  EXPECT_EQ(FilterDataStatus::Continue, filter_->decodeData(req_data, true));
```
Comment (tyxia) [inline test/extensions/filters/http/ext_proc/filter_test.cc:726]: nit: Avoid "Yoda-style" comparisons in the new code you added? go/noyoda

Not sure if OSS should strictly follow this pattern though

**Review by tyxia — APPROVED** [review]
LGTM, modulo nits

**Comment by yanjunxiang-google** [comment]
@envoyproxy/api-shepherds  The API change is just removed a [:not-implemented-hide] comment. PTAL

**Review by htuch — APPROVED** [review]

**Pull request merged**